src/memray/_memray.pyx

```diff
@@ -170,12 +170,21 @@ cdef class AllocationRecord:
 
     def stack_trace(self, max_stacks=None):
         assert self._reader.get() != NULL, "Cannot get stack trace without reader."
+        cdef ssize_t to_skip
+        cdef ssize_t to_keep
         if self._stack_trace is None:
             if self.allocator in (AllocatorType.FREE, AllocatorType.MUNMAP):
                 raise NotImplementedError("Stack traces for deallocations aren't captured.")
 
             if max_stacks is None:
                 self._stack_trace = self._reader.get().Py_GetStackFrame(self._tuple[4])
+                if self._tuple[0] == self._reader.get().getMainThreadTid():
+                    to_skip = self._reader.get().getSkippedFramesOnMainThread()
+                    if len(self._stack_trace) > to_skip:
+                        to_keep = len(self._stack_trace) - to_skip
+                    else:
+                        to_keep = 0
```

Comment:
```suggestion
                    to_keep = max(len(self._track_trace) - to_skip, 0)
```

Reply:
Although this may convert back and forth to Python stuff depending on what Cython decides to do, so I am not sure is a clear win.

Reply:
Ah, I originally had `to_skip` and `to_keep` as `size_t` instead of `ssize_t`, and needed to avoid integer underflow - that's why it was slightly awkward. I'll check the generated code and make sure Cython does something reasonable with this `max` call, and use it if so.